### PR TITLE
Pipeline, Menyalshikov Timur - 21FPL1

### PR DESCRIPTION
**Pull request by Timur20033**

### CODE REVIEW

File: lab_6_pipeline/pipeline.py [hunk 1]
```diff
@@ -18,21 +38,50 @@ def __init__(self, path_to_raw_txt_data: Path):
...
         """
         Validates folder with assets
         """
+        if not self.path_to_raw_txt_data.exists():
+            raise FileNotFoundError
+
+        if not self.path_to_raw_txt_data.is_dir():
+            raise NotADirectoryError
+
+        if len(list(self.path_to_raw_txt_data.iterdir())) == 0:
```
Comment (marina-kaz) [inline lab_6_pipeline/pipeline.py]: `== 0` is a bad practice

File: lab_6_pipeline/pipeline.py [hunk 2]
```diff
@@ -18,21 +38,50 @@ def __init__(self, path_to_raw_txt_data: Path):
...
+            raise EmptyDirectoryError
+
+        texts = [text for text in self.path_to_raw_txt_data.glob(r'*_raw.txt')]
+        text_sizes = [os.path.getsize(text) for text in texts]
+
+        metas = [meta for meta in self.path_to_raw_txt_data.glob(r'*_meta.json')]
+        meta_sizes = [os.path.getsize(text) for text in metas]
+
+        if len(texts) != len(metas):
+            raise InconsistentDatasetError
```
Comment (marina-kaz) [inline lab_6_pipeline/pipeline.py:64]: this error should also be thrown if there is a problem with numeration

File: lab_6_pipeline/pipeline.py [hunk 3]
```diff
@@ -18,21 +38,50 @@ def __init__(self, path_to_raw_txt_data: Path):
...
+
+        if 0 in text_sizes or 0 in meta_sizes:
+            raise InconsistentDatasetError
 
     def _scan_dataset(self) -> None:
         """
         Register each dataset entry
         """
+        for text in self.path_to_raw_txt_data.glob('*.txt'):
+            for index in range(999):
```
Comment (marina-kaz) [inline lab_6_pipeline/pipeline.py]: this is very creative!
however, try to find a more elegant way of determining the index of the article. consider extracting it from the fime name or from the article instance

**Review by marina-kaz — COMMENTED** [review]
Good start!

Keep going, I believe you can get a high mark.

File: lab_6_pipeline/pipeline.py (hunk 3 above)
Comment (Timur20033) [reply lab_6_pipeline/pipeline.py]: fixed

File: lab_6_pipeline/pipeline.py (hunk 2 above)
Comment (Timur20033) [reply lab_6_pipeline/pipeline.py:64]: fixed

File: lab_6_pipeline/pipeline.py (hunk 1 above)
Comment (Timur20033) [reply lab_6_pipeline/pipeline.py]: fixed

File: lab_6_pipeline/pipeline.py [hunk 4]
```diff
@@ -144,16 +211,27 @@ def __init__(self, corpus_manager: CorpusManager):
...
         Initializes MorphologicalAnalysisPipeline
         """
+        self._corpus = corpus_manager
 
     def _process(self, text: str) -> List[ConlluSentence]:
         """
         Returns the text representation as the list of ConlluSentence
         """
+        conllu_sentences = []
+        for sent_idx, sent in enumerate(split_by_sentence(text), start=1):
```
Comment (marina-kaz) [inline lab_6_pipeline/pipeline.py:221]: I believe sentence numeration is supposed to start from 0

**Review by marina-kaz — COMMENTED** [review]
It is good for mark 4!

**Pull request closed without merging**